### PR TITLE
Replace token widget with Erc20 Widget

### PR DESCRIPTION
**Pull request by acolytec3**
Replaces PR #307

### CODE REVIEW

File: components/Investment/TrancheMetric/index.tsx [hunk 1]
```diff
@@ -1,48 +1,63 @@
...
+      }
+    };
+    const tinToken = {
+      [tinAddress] : {
+        symbol: 'TIN',
+        logo: '../../static/TIN_final.svg',
+        decimals: 18,
+        name: 'TIN'
+      }
+    };
```
Comment (ilinzweilin) [inline components/Investment/TrancheMetric/index.tsx:39]: I think the InvestorView Container should take care of retrieving all this informations and making the distinction between junior & senior infos based on the tab selected.
you can extend the tranche object by all the required values and pass it down to this component from the investorView container.
So, this component will just display all the infos that it gets without even bothering if it is dealing with junior or senior data. 

If you get any type issues when extending the  tranche object, please just use any. I will fix it after the pr is merged.


Then you also would not need to add tinlake to the props.

File: components/Investment/TrancheMetric/index.tsx [hunk 2]
```diff
@@ -1,48 +1,63 @@
...
     return <Box margin="none">
       <Box>
         <Heading level="4" margin={{ bottom: 'medium' }}>Investment overview</Heading>
         <Box direction="row" >
-          <Box basis={'1/3'} gap="medium">
             <DashboardMetric label="Investor token balance">
-              <NumberDisplay value={tokenBalance ? baseToDisplay(tokenBalance, 18) : ''} suffix={currencyLabel} precision={18} />
+              {token === 'DROP' && <Erc20Widget value={tokenBalance ? tokenBalance.toString() : '0'} tokenData={dropToken} precision={18}  />}
```
Comment (ilinzweilin) [inline components/Investment/TrancheMetric/index.tsx:45]: If you make the distinction in the investorView container based you won't need to do it here

File: components/Investment/TrancheMetric/index.tsx [hunk 3]
```diff
@@ -1,48 +1,63 @@
...
-          <Box basis={'1/3'} gap="medium">
             <DashboardMetric label="Investment limit">
-              <NumberDisplay value={maxSupply ? baseToDisplay(maxSupply, 18) : ''} suffix=" DAI" precision={18} />
+            <Erc20Widget value={maxSupply ? maxSupply.toString() : '0'} tokenData={DAI} precision={18} />
             </DashboardMetric>
-          </Box>
-          <Box basis={'1/3'} gap="medium">
             <DashboardMetric label="Redeem limit">
-              <NumberDisplay value={maxRedeem ? baseToDisplay(maxRedeem, 18) : ''} suffix={currencyLabel} precision={18} />
+              {token === 'DROP' && <Erc20Widget value={maxRedeem ? maxRedeem.toString() : '0'} tokenData={dropToken} precision={18} />}
```
Comment (ilinzweilin) [inline components/Investment/TrancheMetric/index.tsx:58]: same :)

File: containers/Investment/Tranche/index.tsx [hunk 4]
```diff
@@ -50,7 +50,7 @@ class TrancheView extends React.Component<Props> {
         <Box>
           <Box margin={{ top: 'medium', bottom: 'large' }} >
             <Box>
-              <TrancheMetric tranche={tranche} investor={investor} type={tranche.type as any as TrancheType} />
+              <TrancheMetric tinlake={tinlake} tranche={tranche} investor={investor} type={tranche.type as any as TrancheType} />
```
Comment (ilinzweilin) [inline containers/Investment/Tranche/index.tsx:53]: don't need to pass tinlake as a prop, if extending the tranche object with req token infos

File: containers/Investment/Supply/index.tsx [hunk 5]
```diff
@@ -62,24 +63,12 @@ class InvestorSupply extends React.Component<Props, State> {
     const canSupply = maxSupplyAmount.toString() !== '0' && !maxSupplyOverflow;
     return <Box basis={'1/4'} gap="medium" margin={{ right: 'large' }}>
       <Box gap="medium">
-        <FormField label="Investment amount">
-          <NumberInput value={baseToDisplay(supplyAmount, 18)} suffix=" DAI" precision={18}
-            onValueChange={({ value }) =>
-              this.setState({ supplyAmount: displayToBase(value, 18) })}
-          />
-        </FormField>
+        <Erc20Widget fieldLabel="Redeem token" limit={maxSupplyAmount.toString()} tokenData={DAI} precision={18} onValueChanged={(value : string) =>
```
Comment (ilinzweilin) [inline containers/Investment/Supply/index.tsx:66]: UI is currently not reacting to onValueChange events

File: containers/Investment/Redeem/index.tsx [hunk 6]
```diff
@@ -60,28 +59,36 @@ class InvestorRedeem extends React.Component<Props, State> {
     const availableTokensOverflow = (new BN(redeemAmount).cmp(new BN(tokenBalance)) > 0);
     const redeemEnabled = redeemLimitSet && !limitOverflow && !availableTokensOverflow;
 
+    const dropAddress = this.props.tinlake.contractAddresses.SENIOR_TOKEN as string;
```
Comment (ilinzweilin) [inline containers/Investment/Redeem/index.tsx:62]: here the same: please add this information to the investorView Container and pass it down

File: containers/Investment/Redeem/index.tsx [hunk 7]
```diff
@@ -60,28 +59,36 @@ class InvestorRedeem extends React.Component<Props, State> {
...
-        <FormField label="Redeem token">
-          <NumberInput value={baseToDisplay(redeemAmount, 18)} suffix={` ${tranche.token}`} precision={18}
-            onValueChange={({ value }) =>
-              this.setState({ redeemAmount: displayToBase(value, 18) })}
-          />
-        </FormField>
+        {tranche.type === 'senior' && <Erc20Widget fieldLabel="Redeem token" limit={maxRedeemAmount.toString()} tokenData={dropToken} precision={18} onValueChanged={(value : string) =>
+                this.setState({ redeemAmount: displayToBase(value, 18) })}
+                errorMessage="Max redeem amount exceeded" />}
+        {tranche.type === 'junior' && <Erc20Widget fieldLabel="Redeem token" limit={maxRedeemAmount.toString()} tokenData={tinToken} precision={18} onValueChanged={(value : string) =>
```
Comment (ilinzweilin) [inline containers/Investment/Redeem/index.tsx:85]: UI not reacting to value changes

File: containers/Investment/Supply/index.tsx (hunk 5 above)
Comment (ilinzweilin) [inline containers/Investment/Supply/index.tsx:66]: `fieldLabel="Invest currency"`

**Comment by lucasvo** [comment]
One thing I noticed is that the hover over the (i) icon should make it black (see: https://www.figma.com/file/hXkBIpynGQX4AYCNhpskZD/Axis-Styleguide?node-id=40%3A287)

File: containers/Investment/Redeem/index.tsx [hunk 8]
```diff
@@ -60,28 +59,36 @@ class InvestorRedeem extends React.Component<Props, State> {
...
+    };
     return <Box basis={'1/4'} gap="medium" margin={{ right: 'large' }}>
       <Box gap="medium">
-        <FormField label="Redeem token">
-          <NumberInput value={baseToDisplay(redeemAmount, 18)} suffix={` ${tranche.token}`} precision={18}
-            onValueChange={({ value }) =>
-              this.setState({ redeemAmount: displayToBase(value, 18) })}
-          />
-        </FormField>
+        {tranche.type === 'senior' && <Erc20Widget fieldLabel="Redeem token" limit={maxRedeemAmount.toString()} tokenData={dropToken} precision={18} onValueChanged={(value : string) =>
```
Comment (ilinzweilin) [inline containers/Investment/Redeem/index.tsx:82]: limit should be `min(masRedeemAmount, tokenBalance)`

**Pull request closed without merging**